test_quality_of_service/include/test_quality_of_service/qos_utilities.hpp

```diff
@@ -0,0 +1,118 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef TEST_QUALITY_OF_SERVICE__QOS_UTILITIES_HPP_
+#define TEST_QUALITY_OF_SERVICE__QOS_UTILITIES_HPP_
+
+
+#include <chrono>
+#include <memory>
+#include <mutex>
+#include <string>
+#include <tuple>
+
+#include "gtest/gtest.h"
+
+#include "rclcpp/rclcpp.hpp"
+#include "std_msgs/msg/string.hpp"
+
+
+/// Helper time conversion method
+/**
+ * @param milliseconds
+ * @return tuple of milliseconds converted to <seconds, nanoseconds>
+ */
+std::tuple<size_t, size_t> chrono_milliseconds_to_size_t(
+  const std::chrono::milliseconds & milliseconds);
+
+/// Simple publishing node used for system tests
+class Publisher : public rclcpp::Node
```

Comment:
A namespace or prefixed naming would be nice for these - so that there's no possible confusion with the `rclcpp` types. I'd recommend something along the lines of `QoSTestPublisherNode`

Reply:
Will rename.